### PR TITLE
Update graphql-java-servlet to support graphql-java 16.

### LINKED ISSUE #280: Graphql-Java 16.0 support
Hello, 
A new version of graphql-java was released yesterday [v16.0](https://github.com/graphql-java/graphql-java/releases/tag/v16.0), so it will be cool to profit from latest improvements.
Thanks

### PR DESCRIPTION
**Pull request by artem-ag**
Fixes #280

This is a breaking change that makes library incompatible with prior versions of graphql-java:
- `@defer` directive support is gone.
- `Query` type has to have at least 1 field defined for the schema to be valid.

### CODE REVIEW

**Comment by artem-ag** [comment]
@oliemansm Could you please advise on why Mac OS run fail and whether it's ok? Seem transient, but I can't repro locally. Also, could you please review this PR?

**Comment by oliemansm** [comment]
@BlasiusSecundus Do you have any idea why builds sometimes fail and sometimes succeed by any chance? Can't find any clear cause for this from the logs, and often it's MacOS that's failing.

**Comment by BlasiusSecundus** [comment]
If it is consistently macOS only, possibly this is the cause: https://github.com/actions/virtual-environments/issues/2247

**Comment by oliemansm** [comment]
Seem to be quite a lot of users having this problem: https://github.com/actions/virtual-environments/issues/841.

File: gradle.properties [hunk 1]
```diff
@@ -7,8 +7,9 @@ PROJECT_LICENSE=MIT
 PROJECT_LICENSE_URL=https://github.com/graphql-java-kickstart/spring-boot-graphql/blob/master/LICENSE.md
 PROJECT_DEV_ID=apottere
 PROJECT_DEV_NAME=Andrew Potter
-LIB_GRAPHQL_JAVA_VER=15.0
-LIB_JACKSON_VER=2.10.0
+LIB_GRAPHQL_JAVA_VER=16.0
```
Comment (oliemansm) [inline gradle.properties:10]: There was a particular bug in version 16.0. Can you upgrade to 16.1 instead?

**Comment by BlasiusSecundus** [comment]
I'd say it is best to wait until GitHub finishes the macOS maintenance (should be today sometimes), then try again.

**Review by oliemansm — CHANGES_REQUESTED** [review]
Looks good overall. The only change is to upgrade to 16.1 instead of 16.0, because of a bug in that version. See also: https://github.com/graphql-java/graphql-java/releases/tag/v16.0.

Once that's changed I'll first release a bugfix version of this library supporting graphql-java 15.0 and will immediately after release this one. Hopefully I'm able to do all this early next week.

**Review by oliemansm — APPROVED** [review]

**Pull request merged**